lib/os_io.c

```diff
@@ -801,21 +991,29 @@ ___time timeout;)
 #ifdef USE_select_or_poll
 
 #ifdef USE_select
-
-  state.highest_fd_plus_1 = 0;
-
-  ___FD_ZERO(&state.readfds);
-  ___FD_ZERO(&state.writefds);
-  ___FD_ZERO(&state.exceptfds);
-
+  {
+    ___processor_state ps = ___PSTATE;
```

Comment:
```ps``` should be renamed ```___ps``` (that way if ever code is added that uses a macro accessing ```___ps``` freely, it will work).